### PR TITLE
Build nim in parallel

### PR DESCRIPTION
**Pull request by arnetheduck**
because when does this _not_ make sense?

### CODE REVIEW

File: scripts/build_nim.sh [hunk 1]
```diff
@@ -26,7 +26,8 @@ CI_CACHE="$4"
 [[ -z "$V" ]] && V=0
 [[ -z "$CC" ]] && CC="gcc"
 # to build csources in parallel, set MAKE="make -jN"
-[[ -z "$MAKE" ]] && MAKE="make"
+# the default is 3 processes which is better than 1 but still not too taxing
+[[ -z "$MAKE" ]] && MAKE="make -j3"
```
Comment (stefantalpalaru) [inline scripts/build_nim.sh:30]: Where is this supposed to kick in? We set the "MAKE" env var in the "build-nim" target: https://github.com/status-im/nimbus-build-system/blob/92e5042667b747d22106f085eaa9b5e9766ba474/makefiles/targets.mk#L60

and in CI, when using this script directly: https://github.com/status-im/nim-libp2p/blob/7b5259dbc713851f5fd14d2024fedf2cee4f95c7/azure-pipelines.yml#L102

**Comment by arnetheduck** [comment]
I dunno, but I don't think I've ever seen it being built in parallel:

```
dockrem+  434269  0.0  0.0   1212   744 ?        S    11:12   0:00         make update
dockrem+  438308  0.0  0.0   1212   716 ?        S    11:14   0:00           make --no-print-directory deps
dockrem+  438679  0.0  0.0   1212   700 ?        S    11:14   0:00             make --no-print-directory build-nim
dockrem+  438684  0.0  0.0   2308  1732 ?        S    11:14   0:00               bash /opt/nimbus-eth2/vendor/nimbus-build-system/scripts/build_nim.sh vendor/nimbus-build-system/vendor/Nim ../Nim-csources ../nimble
dockrem+  438702  0.0  0.0   1576  1204 ?        S    11:14   0:00                 make LD=gcc
dockrem+  439232  0.0  0.0   1944     4 ?        S    11:17   0:00                   gcc -Ic_code -w -O3 -fno-strict-aliasing -c -o c_code/2_2/sem.nim.o c_code/2_2/sem.nim.c
dockrem+  439233 90.2  3.4 163924 136864 ?       R    11:17   0:03                     /usr/libexec/gcc/x86_64-alpine-linux-musl/9.3.0/cc1 -quiet -I c_code c_code/2_2/sem.nim.c -quiet -dumpbase sem.nim.c -mtune=generic -march=x86-64 -auxbase-strip c_code/2_2/sem.nim.o -O3 -w -fno-strict-aliasing -o /tmp/ccaAkCIL.s
```

**Comment by arnetheduck** [comment]
maybe there should be a `MAKEFLAGS`  instead

**Comment by arnetheduck** [comment]
ideally we would compile nim itself with LTO as well - there are significant gains to be had on compile speed

**Comment by stefantalpalaru** [comment]
You're supposed to `make -j8 update` or equivalent.

**Pull request closed without merging**